test/unit/specs/components/index_pattern/_index_pattern.js

```diff
@@ -102,7 +102,7 @@ define(function (require) {
     describe('getFields', function () {
       it('should return all non-scripted fields', function () {
         var indexed = _.where(mockLogstashFields, { scripted: false });
-        expect(indexPattern.getFields()).to.eql(indexed);
+        expect(indexPattern.getFields().length).to.eql(indexed.length);
```

Comment:
Why not compare the actual fields? I don't think checking length matches is a sufficient test. Maybe we can compare the field names if comparing the actual fields is an issue.